### ISSUE TITLE
macOS : cannot listen on the TCP port: listen tcp4 :53: bind: address already in us

### ISSUE THREAD

**Issue by Constantin1489** [containers/podman#17690]
### Issue Description

I ran the code below. ( I did `podman machine set --rootful`)

```
$ podman volume create pihole_pihole
$ podman volume create pihole_dnsmasq
$ podman run -d --name pihole7 \
    -e WEBPASSWORD="11111" \
    -e DNS1=8.8.8.8 \
    -e DNS2=1.1.1.1 \
    -v pihole_pihole:/etc/pihole:Z \
    -v pihole_dnsmasq:/etc/dnsmasq.d:Z \
    -p 8889:80 \
    -p 53:53/tcp \
    -p 53:53/udp \
    -p 443:443 pihole/pihole:latest
```

Output : 

```
Error: unable to start container "13f722fc3600bcf6ac50c595dc214fa74e0804b56efe69dcafb7bec67b528f61": cannot listen on the TCP port: listen tcp4 :53: bind: address already in use
```

`$ sudo lsof -i:53`

There is no output.

`$ podman machine info`

Output : 
```
Host:
  Arch: amd64
  CurrentMachine: ""
  DefaultMachine: ""
  EventsDir: /var/folders/_r/c7ygdkk13254ngxz6j7c6y3m0000gn/T/podman-run--1/podman
  MachineConfigDir: /Users/username/.config/containers/podman/machine/qemu
  MachineImageDir: /Users/username/.local/share/containers/podman/machine/qemu
  MachineState: ""
  NumberOfMachines: 1
  OS: darwin
  VMType: qemu
Version:
  APIVersion: 4.4.2
  Built: 1677167961
  BuiltTime: Fri Feb 24 00:59:21 2023
  GitCommit: 74afe26887f814d1c39925a1624851ef3590e79c
  GoVersion: go1.20.1
  Os: darwin
  OsArch: darwin/amd64
```

`$ system_profiler SPSoftwareDataType`

```
Software:

    System Software Overview:

      System Version: macOS 13.2.1 (22D68)
      Kernel Version: Darwin 22.3.0
      Boot Volume: Macintosh HD
      Boot Mode: Normal
      Computer Name:  username
      User Name: username
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
      Time since boot: 1 hour, 49 minutes
```

### Steps to reproduce the issue

Steps to reproduce the issue
1. I believe the code I wrote will reproduce the issue.
2. 
3.

### Describe the results you received

If it's not podman's bug, then it is a macOS config problem.

### Describe the results you expected

The docker image worked great in docker. So I want to know whether I didn't configure ports or it's just a bug.

### podman info output

```yaml
Host:
  Arch: amd64
  CurrentMachine: ""
  DefaultMachine: ""
  EventsDir: /var/folders/_r/c7ygdkk13254ngxz6j7c6y3m0000gn/T/podman-run--1/podman
  MachineConfigDir: /Users/username/.config/containers/podman/machine/qemu
  MachineImageDir: /Users/username/.local/share/containers/podman/machine/qemu
  MachineState: ""
  NumberOfMachines: 1
  OS: darwin
  VMType: qemu
Version:
  APIVersion: 4.4.2
  Built: 1677167961
  BuiltTime: Fri Feb 24 00:59:21 2023
  GitCommit: 74afe26887f814d1c39925a1624851ef3590e79c
  GoVersion: go1.20.1
  Os: darwin
  OsArch: darwin/amd64
```

### Podman in a container

No

### Privileged Or Rootless

Privileged

### Upstream Latest Release

Yes

### Additional environment details

I ran podman on local machine.

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by baude** [comment]
was the lsof run in macos or inside the vm ?

**Comment by Constantin1489** [comment]
@baude, Thank you for helping me.

I ran `lsof` command in macOS. Because of the problem, the container couldn't run on podman. 

When I tried with different container settings, which changed ports, I realized there was no `lsof` in the container.

**the different setting**
```
sudo podman run --rm --name pihole \
    -e WEBPASSWORD="11111" \
    -e DNS1=8.8.8.8 \
    -e DNS2=1.1.1.1 \
    -v pihole_pihole:/etc/pihole:Z \
    -v pihole_dnsmasq:/etc/dnsmasq.d:Z \
    -p 8889:80 \
    -p 8890:53/tcp \
    -p 8890:53/udp \
    -p 8891:443 pihole/pihole:latest
```

**Comment by Luap99** [comment]
Note you need to check in the VM not the container. (`podman machine ssh ss -tulpn`)
I am pretty sure the problem is that inside the VM there is systemd-resolved running on 127.0.0.54:53.
I doubt that we get any benefit from running using systemd-resolved so I suggest that we disable it in the machine VM via ignition.

**Comment by Constantin1489** [comment]
Thank you for giving me perfect answer! I appreciate it! Also you fix my understanding to vm and container!

* [pi-hole README.md : installing on ubuntu or fedora](https://github.com/pi-hole/docker-pi-hole#installing-on-ubuntu-or-fedora)
* podman command using the command from the link above.

```
podman machine ssh sudo sed -r -i.orig 's/#?DNSStubListener=yes/DNSStubListener=no/g' /etc/systemd/resolved.conf
podman machine ssh systemctl restart systemd-resolved
```
